SimPEG/electromagnetics/frequency_domain/survey.py

```diff
@@ -16,13 +16,22 @@ def __init__(self, source_list, **kwargs):
         super(Survey, self).__init__(source_list, **kwargs)
 
         _frequency_dict = {}
+        _source_location_dict = {}
+        _source_location_by_sounding_dict = {}
         for src in self.source_list:
```

Comment:
Not ideal to have a long loop in the init. Probably better to do the operation on-demand only under the property.